src/DictVectors/communicators.jl

```diff
@@ -0,0 +1,306 @@
+struct CommunicatorError <: Exception
+    msg::String
+end
+CommunicatorError(args...) = CommunicatorError(string(args...))
+
+function Base.showerror(io::IO, ex::CommunicatorError)
+    print(io, "CommunicatorError: ", ex.msg)
+end
+
+"""
+    abstract type Communicator
+
+When implementing a communicator, use [`local_segments`](@ref) and
+[`remote_segments`](@ref).
```

Comment:
Can we start with the docstring defining/explaining what a `Communicator` is useful for? Otherwise the uninitiated will just find this thoroughly confusing :-)